src/deploy/functions/build.ts

```diff
@@ -0,0 +1,397 @@
+
+// Trigger definition for arbitrary HTTPS endpoints
+interface HttpsTrigger {
+  // Which service account should be able to trigger this function. No value means "make public
+  // on create and don't do anything on update." For more, see go/cf3-http-access-control
+  invoker?: ServiceAccount | null;
+}
+
+// Trigger definitions for RPCs servers using the HTTP protocol defined at
+// https://firebase.google.com/docs/functions/callable-reference
+// eslint-disable-next-line
+interface CallableTrigger { }
+
+// Trigger definitions for endpoints that should be called as a delegate for other operations.
+// For example, before user login.
+interface BlockingTrigger {
+  eventType: string;
+}
+
+// Trigger definitions for endpoints that listen to CloudEvents emitted by other systems (or legacy
+// Google events for GCF gen 1)
+interface EventTrigger {
+  eventType: string;
+  eventFilters: Record<string, Expression<string>>;
+
+  // whether failed function executions should retry the event execution.
+  // Retries are indefinite, so developers should be sure to add some end condition (e.g. event
+  // age)
+  retry: boolean | Expression<boolean> | null;
+
+  // Region of the EventArc trigger. Must be the same region or multi-region as the event
+  // trigger or be us-central1. All first party triggers (all triggers as of Jan 2022) need not
+  // specify this field because tooling determines the correct value automatically.
+  region?: string | Expression<string>;
+
+  // The service account that EventArc should use to invoke this function. Setting this field
+  // requires the EventArc P4SA to be granted the "ActAs" permission to this service account and
+  // will cause the "invoker" role to be granted to this service account on the endpoint
+  // (Function or Route)
+  serviceAccount?: ServiceAccount | null;
+}
+
+interface TaskQueueRateLimits {
+  maxConcurrentDispatches?: number | Expression<number> | null;
+  maxDispatchesPerSecond?: number | Expression<number> | null;
+}
+
+interface TaskQueueRetryConfig {
+  maxAttempts?: number | Expression<number> | null;
+  maxRetryDurationSeconds: number | Expression<number> | null;
+  minBackoffSeconds?: number | Expression<number> | null;
+  maxBackoffSeconds?: number | Expression<number> | null;
+  maxDoublings?: number | Expression<number> | null;
+}
+
+interface TaskQueueTrigger {
+  rateLimits?: TaskQueueRateLimits | null;
+  retryConfig?: TaskQueueRetryConfig | null;
+
+  // empty array means private
+  invoker?: Array<ServiceAccount | Expression<string>> | null;
+}
+
+interface ScheduleRetryConfig {
+  retryCount?: number | Expression<number> | null;
+  maxRetrySeconds?: number | Expression<number> | null;
+  minBackoffSeconds?: number | Expression<number> | null;
+  maxBackoffSeconds?: number | Expression<number> | null;
+  maxDoublings?: number | Expression<number> | null;
+}
+
+interface ScheduleTrigger {
+  schedule: string | Expression<string>;
+  timeZone: string;
```

Comment:
I wonder if this should be an Expression as well. People could conceivably use a SelectInput

Reply:
I think that's reasonable. Done.